api/tests/test_asset_log.py

```diff
@@ -337,6 +338,109 @@ def test_authenticated_admin_user_get_of_asset_logs_invalid_filter(
         self.assertEqual(response.status_code, 200)
         self.assertEqual(len(response.data["results"]), 0)
 
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_year(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = (
+            f"{self.asset_logs_url}?year={self.test_other_asset.created_at.year}"
+        )
+
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        data = response.data["results"]
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            data[0]["created_at"].split("-")[0],
+            f"{self.test_other_asset.created_at.year}",
+        )
+
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_invalid_year(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
+        asset_logs_url = (
+            f"{self.asset_logs_url}?year=5"
+        )
+
+        response = client.get(
+            asset_logs_url, HTTP_AUTHORIZATION=f"Token {self.token_admin}"
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data["results"]), 0)
+
+    @patch("api.authentication.auth.verify_id_token")
+    def test_filter_asset_logs_by_month(self, mock_verify_id_token):
+        mock_verify_id_token.return_value = {"email": self.admin_user.email}
+        AssetLog.objects.create(
+            checked_by=self.security_user, asset=self.test_other_asset, log_type=CHECKIN
+        )
```

Comment:
You could update the month this asset-log was created and assert that you could filter it out from the rest. Just update the `created_at` attribute.

Reply:
I tried this out and it brought errors that's why I left it out